pagination/input.js

```diff
@@ -12,217 +12,67 @@
-					iNewStart = 0;
-				}
-				if (iNewStart >= oSettings.fnRecordsDisplay()) {
-					iNewStart = (Math.ceil((oSettings.fnRecordsDisplay()) / oSettings._iDisplayLength) - 1) * oSettings._iDisplayLength;
-				}
-
-				oSettings._iDisplayStart = iNewStart;
-				oSettings.oInstance.trigger("page.dt", oSettings);
-				fnCallbackDraw(oSettings);
-			});
-
-			// Take the brutal approach to cancelling text selection.
-			$('span', nPaging).bind('mousedown', function () { return false; });
-			$('span', nPaging).bind('selectstart', function() { return false; });
-
-			// If we can't page anyway, might as well not show it.
-			var iPages = calcPages(oSettings);
-			if (iPages <= 1) {
-				$(nPaging).hide();
-			}
-		},
-
-		'fnUpdate': function (oSettings) {
-			if (!oSettings.aanFeatures.p) {
-				return;
-			}
-
-			var iPages = calcPages(oSettings);
-			var iCurrentPage = calcCurrentPage(oSettings);
-
-			var an = oSettings.aanFeatures.p;
-			if (iPages <= 1) // hide paging when we can't page
-			{
-				$(an).hide();
-				return;
-			}
-
-			var disableClasses = calcDisableClasses(oSettings);
-
-			$(an).show();
-
-			// Enable/Disable `first` button.
-			$(an).children('.' + firstClassName)
-				.removeClass(oSettings.oClasses.sPageButtonDisabled)
-				.addClass(disableClasses[firstClassName]);
-
-			// Enable/Disable `prev` button.
-			$(an).children('.' + previousClassName)
-				.removeClass(oSettings.oClasses.sPageButtonDisabled)
-				.addClass(disableClasses[previousClassName]);
-
-			// Enable/Disable `next` button.
-			$(an).children('.' + nextClassName)
-				.removeClass(oSettings.oClasses.sPageButtonDisabled)
-				.addClass(disableClasses[nextClassName]);
-
-			// Enable/Disable `last` button.
-			$(an).children('.' + lastClassName)
-				.removeClass(oSettings.oClasses.sPageButtonDisabled)
-				.addClass(disableClasses[lastClassName]);
-
-			// Paginate of N pages text
-			$(an).find('.' + paginateTotalClassName).html(iPages);
-
-			// Current page number input value
-			$(an).find('.' + paginateInputClassName).val(iCurrentPage);
-		}
-	};
+    function createInputElement(settings) {
+        const pageInfo = $(settings.nTable).DataTable().page.info();
+        const $input = $('<input>', {
+            class: 'paginate_input',
+            type: 'number',
+            min: 1,
+            max: pageInfo.pages,
+        }).val(pageInfo.page + 1);
+        $input.on("change", function (e) {
+            if (this.value === '' || this.value.match(/[^0-9]/)) {
+                /* Nothing entered or non-numeric character */
+                this.value = this.value.replace(/[^\d]/g, ''); // don't even allow anything but digits
+                return;
+            }
+
+            const page = Number(this.value - 1);
+            $(settings.nTable).DataTable().page(page).draw(false);
+        });
+
+        return $input;
+    }
+
+    $.fn.DataTable.ext.pager.input = function() {
+        return ['first', 'previous', 'ellipsis', 'next', 'last'];
```

Comment:
I had to use `ellipsis` because it complains and logs errors if you provide a non-builtin value.